python/pyspark/sql/datasource.py

```diff
@@ -298,6 +305,90 @@ def read(self, partition: InputPartition) -> Iterator[Union[Tuple, Row]]:
         ...
 
 
+class DataSourceStreamReader(ABC):
+    def initial_offset(self) -> dict:
```

Comment:
I think the naming should be `initialOffset`.

Or ... you could just have one method called `offset`, and returns a tuple or a Data Class that has `initial` and `latest` property to make this more Pythonic.

Reply:
Changed to camel case